content/posts/2022-12-24-react-native-redux-toolkit/index.md

```diff
@@ -0,0 +1,503 @@
+
+### Install redux-persist
+
+Install `redux-persist` using the following command:
+
+```bash
+npm install redux-persist
+```
+
+We will also need to install `@react-native-async-storage/async-storage` to use `AsyncStorage` as our storage engine.
+
+```bash
+npx expo install @react-native-async-storage/async-storage
+```
+
+### Create a persistor
+
+Go to `store.js`. We will do some modifications to set up our redux store with redux-persist.
+
+```tsx
+import { configureStore } from '@reduxjs/toolkit';
+import { persistStore, persistReducer } from 'redux-persist';
+
+// adding our persist configs
+const persistConfig = {
+  key: 'root',
+  version: 1,
+  storage: AsyncStorage,
+};
+
+// adding our rootReducer
+const rootReducer = combineReducers({
+  balance: balanceReducer,
+});
+
+// persisting our rootReducer
+const persistedReducer = persistReducer(persistConfig, rootReducer);
+
+// creating our store and exporting it
+export const store = configureStore({
+  reducer: persistedReducer,
+});
+```
+
+Once our store is created, we need to create a persistor and provide it to our store. Let's go to `App.tsx` and add the following code:
+
+```tsx
+........
+........
+import { persistStore } from "redux-persist";
+import { PersistGate } from "redux-persist/integration/react";
+import { store } from "./store";
+
+let persistor = persistStore(store);
+
+export default function App() {
+  ......
+  ......
+  if (!isLoadingComplete) {
+    return null;
+  } else {
+    return (
+      <Provider store={store}>
+        <PersistGate loading={null} persistor={persistor}>
+          .....
+          .....
+        </PersistGate>
+      </Provider>
+    );
+  }
+}
+```
+
+We have imported `persistStore` and `PersistGate` from `redux-persist`. We have also imported our store from `store.js`. We have created a persistor using the `persistStore` function and provided it to the `PersistGate` component. The `PersistGate` component will wait until the store is persisted and then it will render the app.
+
+To test if our balance state is persisted, deposit or withdraw some amount. Close the app and open it again. You will see that the balance state is persisted and the app is rendered with the persisted balance state.
+
+## Conclusion
+
+In this tutorial, we have learned how to use Redux Toolkit to manage the state of our React Native app. We have also learned how to persist our redux store using redux-persist. At the same time we have also learned how to use the React Native Debugger to debug our React Native app, specifically our Redux store. I hope you enjoyed this tutorial!
+
+## Source Code
+
+You can find the source code for this tutorial on [GitHub](https://github.com/notJust-dev/react-native-redux-toolkit)
```

Comment:
Is it possible to add a snack as well?

Reply:
I actually tried this so many times but weirdly I am getting errors inside snack. Somehow it can not find screenone and screentwo, no matter what I do.